osu.Game/Overlays/BeatmapSetOverlay.cs

```diff
@@ -124,7 +127,8 @@ protected override void PopIn()
         protected override void PopOut()
         {
             base.PopOut();
-            header.Details.StopPreview();
+            if (previewTrackManager.CurrentTrack?.Owner == this)
+                previewTrackManager.CurrentTrack?.Stop();
```

Comment:
I think you'd be best off making the `Owner` private and passing in `this` to the `Stop()` method as an optional parameter

`void Stop(OverlayContainer source = null)`

Make sure to add xmldoc for this explaining what the source is used for.